sdk/dotnet/Pulumi/Pulumi.csproj

```diff
@@ -24,6 +24,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="OneOf" Version="2.1.151" />
```

Comment:
Does this library give much value to take a dependency on it in our public API surface? We could add `OneOf` (or `Either`, `Choice`, etc) as our own type.

Reply:
I had the same question actually - it's unfortunate that this dependency ends up getting pushed into user code as well.  It does seem like it's likely worth just owning our own type here to avoid users having to think about this.